core/src/test/scala/unit/kafka/server/ShareGroupHeartbeatRequestTest.scala

```diff
@@ -465,7 +467,8 @@ class ShareGroupHeartbeatRequestTest(cluster: ClusterInstance) {
         shareGroupHeartbeatResponse.data.assignment.topicPartitions.containsAll(expectedAssignment.topicPartitions)
     }, msg = s"Could not get partitions for topic foo and bar assigned. Last response $shareGroupHeartbeatResponse.")
     // Verify the response.
-    assertEquals(3, shareGroupHeartbeatResponse.data.memberEpoch)
```

Comment:
The root cause is that the broker can complete all topic assignments if it has the latest metadata. Otherwise, it requires an extra cycle to finish. We should add a metadata sync before starting the loop.